lib/src/internal/shard/shard_handler.dart

```diff
@@ -59,21 +63,29 @@ class ShardRunner implements Disposable {
   /// Handler for uncompressed messages received from Discord.
   ///
   /// Calls jsonDecode and sends the data back to the manager.
-  void receive(String payload) => execute(ShardMessage(
-        ShardToManager.received,
-        seq: seq--,
-        data: jsonDecode(payload),
-      ));
+  void receive(/* List<int>|RawApiMap|String */ dynamic payload) {
+    if (payload is List<int>) {
+      payload = utf8.decode(zlib.decode(payload));
```

Comment:
Bit of a nit but we should either move the decompression somewhere else or update the doc comment to reflect that the data may still be compressed.

Reply:
That now for now I think since its PR for release. We can do bug fix later for that